ddlitlab2024/dataset/recording2mcap.py

```diff
@@ -0,0 +1,308 @@
+import json
+import shutil
+import sys
+from pathlib import Path
+
+from sqlalchemy.orm import Session
+
+from ddlitlab2024.dataset import logger
+from ddlitlab2024.dataset.models import Recording, stamp_to_seconds_nanoseconds
+
+try:
+    import rosbag2_py
+    from builtin_interfaces.msg import Time
+    from geometry_msgs.msg import Quaternion
+    from rclpy.serialization import serialize_message
+    from sensor_msgs.msg import Image, JointState
+    from std_msgs.msg import Header, String
+except ImportError:
+    logger.error(
+        "Failed to import ROS 2 packages. These are necessary to convert recordings to .mcap files. "
+        "Make sure ROS 2 installed and sourced."
+    )
+    sys.exit(1)
+
+
+def get_recording(db: Session, recording_id_or_filename: str | int) -> Recording:
+    """Get the recording from the input string or integer
+
+    param db: The database
+    param recording_id_or_filename: The recording ID or original filename
+    raises ValueError: If the recording does not exist
+    return: The recording
+    """
+    if isinstance(recording_id_or_filename, int) or recording_id_or_filename.isdigit():
+        # Verify that the recording exists
+        recording_id = int(recording_id_or_filename)
+        recording = db.query(Recording).get(recording_id)
+        if recording is None:
+            raise ValueError(f"Recording '{recording_id}' not found")
+        return recording
+    elif isinstance(recording_id_or_filename, str):
+        recording = db.query(Recording).filter(Recording.original_file == recording_id_or_filename).first()
+        if recording is None:
+            raise ValueError(f"Recording with original filename '{recording_id_or_filename}' not found")
+        return recording
+    else:
+        raise TypeError("Recording ID must be an integer or string")
+
+
+def get_writer(output: Path) -> rosbag2_py.SequentialWriter:
+    """Get the mcap writer.
+
+    param output: The output mcap file
+    return: The mcap writer
+    """
+    if output.exists():
+        # Ask the user if they want to overwrite the existing file
+        if not input(f"Output directory '{output}' already exists. Overwrite? (y/n): ").lower().startswith("y"):
+            logger.info("Exiting")
+            sys.exit(0)
+        # Remove the existing directory
+        shutil.rmtree(output)
+
+    writer = rosbag2_py.SequentialWriter()
+    writer.open(
+        rosbag2_py.StorageOptions(uri=str(output), storage_id="mcap"),
+        rosbag2_py.ConverterOptions(
+            input_serialization_format="cdr",
+            output_serialization_format="cdr",
+        ),
+    )
+    return writer
+
+
+def write_recording_info(recording: Recording, writer: rosbag2_py.SequentialWriter) -> None:
+    """Write the recording info as a JSON encoded String message to the mcap file
+
+    param recording: The recording
+    param writer: The mcap writer
+    """
+    # Create topic
```

Comment:
I would remove all:
- `# Create topic`
- `# Write ...`

comments in this file